src/cli/tree.rs

```diff
@@ -0,0 +1,396 @@
+            } else {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.down)
+            };
+            print_dependency_leaf(dep, new_prefix, dep_map, visited_pkgs);
+        }
+    }
+}
+
+/// Extract the direct Conda and PyPI dependencies from the environment
+fn direct_dependencies(
+    environment: &crate::project::Environment<'_>,
+    platform: &Platform,
+) -> Vec<String> {
+    let mut project_dependency_names = environment
+        .dependencies(None, Some(*platform))
+        .names()
+        .map(|p| p.as_source().to_string())
+        .collect_vec();
+    project_dependency_names.extend(
+        environment
+            .pypi_dependencies(Some(*platform))
+            .into_iter()
+            .map(|(name, _)| name.as_normalized().to_string()),
+    );
+    project_dependency_names
+}
+
+#[derive(Debug)]
+struct Pkg {
+    name: String,
+    version: String,
+    dependencies: Vec<String>,
+}
+
+#[derive(Debug)]
+struct InvertedPkg {
+    name: String,
+    version: String,
+    needed_by: Vec<String>,
+}
+
+/// Builds a hashmap of dependencies, with names, versions, and what they depend on
+fn generate_dependency_map(locked_deps: &Vec<Package>) -> HashMap<String, Pkg> {
+    let mut dep_map = HashMap::new();
+
+    for dep in locked_deps {
+        let version = dep.version().into_owned();
+
+        if let Some(dep) = dep.as_conda() {
+            let name = dep.package_record().name.as_normalized().to_string();
+            let mut dependencies = Vec::new();
+            for d in dep.package_record().depends.iter() {
+                if let Some((dep_name, _)) = d.split_once(' ') {
+                    dependencies.push(dep_name.to_string())
+                }
+            }
+
+            dep_map.insert(
+                name.clone(),
+                Pkg {
+                    name: name.clone(),
+                    version,
+                    dependencies: unique_deps(dependencies),
+                },
+            );
+        } else if let Some(dep) = dep.as_pypi() {
+            let name = dep.data().package.name.to_string();
+
+            let mut dependencies = Vec::new();
+            for p in dep.data().package.requires_dist.iter() {
+                if let Some(markers) = &p.marker {
+                    tracing::info!(
+                        "A bunch of markers on {}, skipping for now {:?}",
+                        p.name,
+                        markers
+                    );
+                } else {
+                    dependencies.push(p.name.to_string())
+                }
+            }
+            dep_map.insert(
+                name.clone(),
+                Pkg {
+                    name: name.clone(),
+                    version,
+                    dependencies: unique_deps(dependencies),
+                },
+            );
+        }
+    }
+    dep_map
+}
+
+/// Only return the unique dependencies
+fn unique_deps(dependencies: Vec<String>) -> Vec<String> {
```

Comment:
We like to use: https://docs.rs/itertools/0.7.4/itertools/trait.Itertools.html#method.unique

e.g.
```rust
let unique_dependencies = dependencies.into_iter().unique().collect();
```

Reply:
Oh, that's much nicer!